### PR TITLE
populate some more legacy checkpoints

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

adding some more legacy checkpoints, followup on #5223

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/checkpointing/test_legacy_checkpoints.py [hunk 1]
```diff
@@ -24,9 +24,27 @@
...
+    "1.0.3",
+    "1.0.4",
+    "1.0.5",
+    "1.0.6",
+    "1.0.7",
+    "1.0.8",
+    "1.1.0",
+    "1.1.1",
+    "1.1.2",
+    "1.1.3",
```
Comment (justusschock) [inline tests/checkpointing/test_legacy_checkpoints.py:47]: I assume the checkpoint files are already added?

**Review by justusschock — APPROVED** [review]

File: tests/checkpointing/test_legacy_checkpoints.py (hunk 1 above)
Comment (Borda) [reply tests/checkpointing/test_legacy_checkpoints.py:47]: yes, I had to add the old manually, otherwise, tests won't pass ;]

**Review by SkafteNicki — APPROVED** [review]

**Comment by Borda** [comment]
well not sure why the newly added ckpt fails, locally it works fine...

File: tests/checkpointing/test_legacy_checkpoints.py [hunk 2]
```diff
@@ -24,9 +24,27 @@
 CHECKPOINT_EXTENSION = ".ckpt"
 
 
-# todo: add more legacy checkpoints :]
+# todo: add more legacy checkpoints :] for < v0.8
```
Comment (tchaton) [inline tests/checkpointing/test_legacy_checkpoints.py:27]: Can you move this list to utilities and add a test to check current version to it. The test should fail if the current version hasn't been added.

**Review by tchaton — APPROVED** [review]

File: tests/checkpointing/test_legacy_checkpoints.py (hunk 2 above)
Comment (Borda) [reply tests/checkpointing/test_legacy_checkpoints.py:27]: > Can you move this list to utilities

What do you mean, pls?

well, there is nothing such as the current version as current versions are always in development...

**Comment by Borda** [comment]
seems that there is a problem with PT, as the same files can be loaded in 1.5 1.6 1.7 but not in 1.4
and it includes also the later PL versions such as 1.1.3
```
FAILED tests/checkpointing/test_legacy_checkpoints.py::test_resume_legacy_checkpoints[0.8.3]
FAILED tests/checkpointing/test_legacy_checkpoints.py::test_resume_legacy_checkpoints[0.8.4]
FAILED tests/checkpointing/test_legacy_checkpoints.py::test_resume_legacy_checkpoints[0.8.5]
FAILED tests/checkpointing/test_legacy_checkpoints.py::test_resume_legacy_checkpoints[0.9.0]
FAILED tests/checkpointing/test_legacy_checkpoints.py::test_resume_legacy_checkpoints[1.0.8]
FAILED tests/checkpointing/test_legacy_checkpoints.py::test_resume_legacy_checkpoints[1.1.0]
FAILED tests/checkpointing/test_legacy_checkpoints.py::test_resume_legacy_checkpoints[1.1.1]
FAILED tests/checkpointing/test_legacy_checkpoints.py::test_resume_legacy_checkpoints[1.1.2]
FAILED tests/checkpointing/test_legacy_checkpoints.py::test_resume_legacy_checkpoints[1.1.3]
```

File: legacy/zero_training.py [hunk 3]
```diff
@@ -49,6 +49,7 @@ def _loss(self, batch, prediction):
     def _step(self, batch, batch_idx):
         output = self.layer(batch)
         loss = self._loss(batch, output)
+        # return {'loss': loss}
```
Comment (carmocca) [inline legacy/zero_training.py:52]: remove this?

```suggestion
```

**Comment by carmocca** [comment]
> seems that there is a problem with PT, as the same files can be loaded in 1.5 1.6 1.7 but not in 1.4
> and it includes also the later PL versions such as 1.1.3

what's the error?

**Comment by Borda** [comment]
> > seems that there is a problem with PT, as the same files can be loaded in 1.5 1.6 1.7 but not in 1.4 and it includes also the later PL versions such as 1.1.3
> 
> what's the error?

PT somehow states that the ckpt is an archive...
```
                if _is_zipfile(f):
                    # .zip is used for torch.jit.save and will throw an un-pickling error here
>                   raise RuntimeError("{} is a zip archive (did you mean to use torch.jit.load()?)".format(f.name))
E                   RuntimeError: /home/runner/work/pytorch-lightning/pytorch-lightning/legacy/checkpoints/1.1.3/epoch=4-step=319.ckpt is a zip archive (did you mean to use torch.jit.load()?)
```

**Comment by Borda** [comment]
this seems to be very similart issue:
- https://github.com/pytorch/pytorch/issues/25214

so the problem is that we cannot save checkpoints with and load with lover version, the breaking change is in 1.7 so for creating legacy checkpoints we have to use the lowest supported version...

**Comment by Borda** [comment]
ok, seems that there is some problem with `v0.8.5` and `v0.9.0` so I would leave it for another PR...
eventually, we may consider making some workaround to transform archive to loadable format or use `torch.jit`

**Pull request merged**